ember-client/src/main/scala/org/http4s/ember/client/internal/ClientHelpers.scala

```diff
@@ -212,12 +213,15 @@ private[client] object ClientHelpers {
         req: Request[F],
         result: Either[Throwable, Response[F]]): Boolean =
       (req.method.isIdempotent || req.headers.get[`Idempotency-Key`].isDefined) &&
-        isEmptyStreamError(result)
+        isRetryableError(result)
 
-    def isEmptyStreamError[F[_]](result: Either[Throwable, Response[F]]): Boolean =
+    def isRetryableError[F[_]](result: Either[Throwable, Response[F]]): Boolean =
       result match {
         case Right(_) => false
         case Left(_: ClosedChannelException) => true
+        case Left(ex: IOException) =>
+          val msg = ex.getMessage()
+          msg == "Connection reset by peer" || msg == "Broken pipe"
```

Comment:
Are there specific exception classes we can match for, instead of relying on the messages? We've ported many of the exceptions to JS so that the semantics match, but at least right now the messages don't.

https://github.com/typelevel/fs2/blob/main/io/js/src/main/scala/fs2/io/net/NetException.scala

Reply:
Oh, sorry, let me actually read what you wrote above 😅 apologies.

Reply:
Hmm, frustrating. Maybe we should at least _try_ to check for those specific exception classes? And leave this clause as a final catch-all?

Reply:
not in the JDK. classes like `IOException` and `SocketException` are effectively catch-alls for a large space of IO errors across various platforms, and we're only interested in a subset of those

for example, on my mac I get `java.io.IOException: Connection reset by peer`, but i've seen `SocketException` with the same message on some Linux machines

Reply:
> Maybe we should at least try to check for those specific exception classes?

can you elaborate which exceptions would we be looking for ?

Reply:
> can you elaborate which exceptions we would be looking for ?

My bad, you're totally right, there don't seem to be specific exception classes for these. The place to fix this is upstream in fs2.io.js and it's going to be a PITA.